ports/mimxrt/machine_i2c.c

```diff
@@ -50,10 +50,9 @@ typedef struct _machine_i2c_obj_t {
     LPI2C_Type *i2c_inst;
     uint8_t i2c_id;
     uint8_t i2c_hw_id;
-    bool transfer_busy;
-    bool transfer_error;
-    bool nack_flag;
     uint8_t mode;
+    bool transfer_busy;
+    status_t transfer_status;
```

Comment:
Many of the changes in this commit are not related to the force stop mode and it would be better to just move them to the previous commit, which would make less confusion.

Reply:
This part of the code is changed in further commits. So I leave it as it is. And I will definitely not start to split up commits, which will be squashed at the end anyhow. Please make you comment to the final state of the files, not some intermediate states. That is pointless.

Reply:
But how could I know if you intend to squash commits and which of them you intend to squash?

Reply:
Damien will probably squash all commits of this PR into a single one, when he merges the commit. So do not look at individual commits here, because the only purpose of the PR is providing HW I2C.

Reply:
Ach, ok. I find this strategy strange as, well, what should be the commit message of such a squashed commit? :) But, well, its up to Damien. I'll be reviewing the result only from now on.